### PR TITLE
Killing off inconsistencies with impact wrenches / combi-tools

### PR DESCRIPTION
**Pull request by JohnWildkins**
Part 1 in the "why my impact wrench no wrench bolt" saga

tl;dr for end users: combitools / impact wrenches now work on everything their non-powered versions do (except mechs and RIGs, tune in next time on dragon ball Z for that)

essentially all this PR does is murder every istype() check possible that could use a helper function instead, most notably many pen checks with ispen(). it also repaths combitools and powerdrills to /obj/item/weapon so they aren't instantly taken out of contention by half of the known attackby() prompts

I already squashed a couple runtimes here and there from the pen changes and it's possible I missed another because pencode is another scourge upon our lives

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!bugfix

**Comment by JohnWildkins** [comment]
!review
